### ISSUE TITLE
Hi!  What's the best way to run this?

### ISSUE THREAD

**Issue by kenerwin88** [JohnDorsey/PyCellEliminationRun#1]
Hello!  This looks extremely cool.  Just wanted to check, what's the best way to run this?  Which script should I call/and how does it compare to FLAC?  Are you open for contributions / PRs?  Thank you!

**Comment by JohnDorsey** [comment]
Hello Ken!! The process of using it is not very streamlined yet, but I just created a readme with the steps to compress and decompress files. Most tests are done by importing Testing.py first and then it imports the other files.

It seems to compress better than FLAC for small files (75kB / 1-2 seconds of CD quality audio), but I just discovered that .wav.tar.gz also compresses better than FLAC for files that small, so maybe FLAC just has like 8 kilobytes or more of per-file overhead, I don't know, I need to do more testing.

In terms of how it works, CellElimRun is very different from FLAC.
FLAC works by trying to predict each sample based on previous ones and compressing the sequence of errors with deflate, so it is kinda like a 1-dimensional PNG file.
CellElimRun works like a single player game of battleship, where the input audio samples place ships in cells in the ocean, and the codec fires at the cells of the grid in any order. Only the number of misses in a row between hits are stored, and this number codes for the location of exactly one hit, and that location is the index and value of a single sample that's present in the uncompressed sound. At all times in the game, it tries to miss as many times in a row as possible, because the longer the run of cell eliminations goes, the lower the marginal cost of storing another miss, because the cost of storing a number is similar to its logarithm. Whenever a run ends by it accidentally hitting an occupied cell, it learns the index and value of the sample that its prediction model had the least accurate prediction for, which is also some of the most valuable information for improving the accuracy of the model (I think, especially if the model is a fourier transform, but I haven't gotten as far as implementing fourier interpolation yet). Then in the endgame, it has missed so many times that it can't miss anymore, and the last ships are known to be in the only places it hasn't fired yet. So it always describes samples in the order of least predictable to most predictable.
And then I GZIP or LZMA the entire thing because it always helps.

I am open for contributions! but I've never managed a git project with another contributor before, so I don't know much about how.
Thanks!
